runner/call_template_data.go

```diff
@@ -111,13 +112,40 @@ func (td *callTemplateData) executeMetadata(metadata string) (map[string]string,
 			input = tpl.Bytes()
 		}
 
-		err = json.Unmarshal(input, &mdMap)
+		err = json.Unmarshal(input, &md)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	return mdMap, nil
+	return md, nil
+}
+
+// Same as executeMetadata, but this method ensures that the input metadata JSON string is always
+// an array. If the input is an object, but not an array, it's converted to an array.
+func (td *callTemplateData) executeMetadataArray(metadata string) ([]map[string]string, error) {
+	var mdArray []map[string]string
+	var metadataSanitized = strings.TrimSpace(metadata)
+
+	// If the input is an object and not an array, we ensure we always work with an array
+	if !strings.HasPrefix(metadataSanitized, "[") && !strings.HasSuffix(metadataSanitized, "]") {
+		metadata = "[" + metadataSanitized + "]"
+	}
+
+	if len(metadata) > 0 {
+		input := []byte(metadata)
+		tpl, err := td.execute(metadata)
```

Comment:
@bojand I started looking and profiling this code again and I think it would be good to add another optimization here so we don't re-render / evaluate the template and unmarshall the string for every single request.

That's especially important when using large metadata JSON objects like in my case.

Basically, I'm looking at adding some new flag (e.g. ``--plaintext-metadata``, open to better naming) and when this flag is used we don't evaluate metadata as a template and cache it on the worker object and re-use it for subsequent requests, similar as we do with ``w.cachedMessages``.

This should substantially speed things up and reduce memory usage (I'm just testing this change to confirm that).

WDYT?

Reply:
Here we go, here is a quick WIP version - d586c351f637b53f38aece32bc63fd58ce8d7276.

I confirmed that using ``--plaintext-metadata`` flag is much more efficient when working with large metadata JSON objects / arrays (like in my case) and results in  lower worker CPU usage.